scripts/cfFindIP.sh

```diff
@@ -27,21 +42,52 @@ scriptDir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 resultDir="$scriptDir/../result"
 resultFile="$resultDir/$now-result.cf"
 
+#check if expected output folder exists and create if it's not availbe
+if [ ! -d "$resultDir" ]; then
+    mkdir -p "$resultDir"
+fi
+
+
 # Function fncCheckSubnet
 # Check Subnet
 function fncCheckSubnet {
-	local ipList resultFile 
+	local ipList resultFile timeoutCommand
 	ipList="$1"
 	resultFile="$2"
+# Check if 'timeout' command is available
+if command -v timeout >/dev/null 2>&1; then
+    timeoutCommand="timeout"
+else
+    # Check if 'gtimeout' command is available
+    if command -v gtimeout >/dev/null 2>&1; then
+        timeoutCommand="gtimeout"
+    else
+        # Check if 'perl-timeout' command is available
+        if command -v perl-timeout >/dev/null 2>&1; then
+            timeoutCommand="perl-timeout"
+        else
+            # Check if 'python-timeout' command is available
+            if command -v python-timeout >/dev/null 2>&1; then
+                timeoutCommand="python-timeout"
+            else
+                echo >&2 "I require 'timeout' command but it's not installed. Please install 'timeout' or an alternative command like 'gtimeout' or 'perl-timeout' or 'python-timeout' and try again."
+                exit 1
+            fi
+        fi
+    fi
+fi
+
+
 	for ip in ${ipList}
 	do
-		if timeout 1 bash -c "</dev/tcp/$ip/443" > /dev/null 2>&1;
-		then
-			timeMil=$(timeout 2 curl -s -w '%{time_total}\n' --resolve scan.sudoer.net:443:"$ip" https://scan.sudoer.net/data.100K --output /dev/null | xargs -I {} echo "{} * 1000 /1" | bc )
-			if [[ "$timeMil" ]]
+    	if $timeoutCommand 1 nc -z "$ip" 443; then #added nc to support alll comon distros 
```

Comment:
I have another idea
Use sh -c "</dev/tcp/$ip/443" instead of nc -z "$ip" 443
You can find sh everywhere

Reply:
yes it worked!,nc is removed now
Also, I had to remove the ping because it was returning invalid results if the number of parallel threads was too high(lots of 0 seconds which were caused by timeouts), I tried increasing the ping timeout, but it affected the script performance, so I removed it altogether